### PR TITLE
Change the type of entrypoint to [Text]

### PR DESCRIPTION
**Pull request by kseo**
Entrypoint can be an array of strings.

### CODE REVIEW

**Comment by jprider63** [comment]
It looks like `entrypoint` can either be a string, string array, or null. I don't think these changes to `FromJSON ContainerConfig` will work in all cases.

**Comment by kseo** [comment]
The [Docker Remote API](
https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/
) document says that `entrypoint` is either a string or an array of strings, but it seems to return `null` in the examples.

> Entrypoint - Set the entry point for the container as a string or an array of strings.

So, I updated the PR to handle all three cases: 
- null
- a string
- an array of strings

File: src/Docker/Client/Types.hs [hunk 1]
```diff
@@ -1391,7 +1392,9 @@ instance FromJSON ContainerConfig where
         image <- o .: "Image"
         volumes <- o .: "Volumes"
         workingDir <- o .:? "WorkingDir"
-        entrypoint <- o .:? "Entrypoint"
+        entrypoint <- asum [ o .:? "Entrypoint" .!= []
+                           , (:[]) <$> o .: "Entrypoint"
```
Comment (jprider63) [inline src/Docker/Client/Types.hs]: I think this line will always cause parsing to fail when `Entrypoint` isn't a string. Maybe something like the following? 

    (:[]) <$> o .:? "Entrypoint" .!= []

Could you include test cases for parsing each version (null, string, string list)?

Comment (kseo) [inline src/Docker/Client/Types.hs]: Thanks for your comment!

`(:[]) <$> o .:? "Entrypoint" .!= []
`
The code above does not work as it gives rise to `[[Text]]` instead of `[Text]`.

I tested my patch manually and confirmed that it works correctly for all three types. I will create a new PR including test cases!

Comment (denibertovic) [inline src/Docker/Client/Types.hs]: @kseo Adding tests would be great. Thanks! I've pushed changes to my dev branch that fix Travis tests and adds a matrix build for GHC 8.0 so if you could rebase you're changes that would be great.

**Comment by denibertovic** [comment]
This should not have been closed. See here for explanation: #27
